ui/src/app/job-details/tasks/tasks.component.html

```diff
@@ -9,23 +9,36 @@
           <mat-cell *matCellDef="let t" class="task-title">
             <a class="title-link" *ngIf="t.jobId" href="{{ '/jobs/' + t.jobId }}">{{ t.name }}</a>
             <p *ngIf="!t.jobId" class="title-nolink">{{ t.name }}</p>
-            <button *ngIf="isScattered(t)" mat-icon-button class="shard-icon" [matMenuTriggerFor]="shardMenu">
+          </mat-cell>
+        </ng-container>
+
+        <!-- Shard Statuses column -->
+        <ng-container matColumnDef="taskInfoIcons">
+          <mat-header-cell *matHeaderCellDef class="slim-cell"></mat-header-cell>
+          <mat-cell *matCellDef="let t" class="slim-cell">
+            <button *ngIf="t.shardStatuses" mat-icon-button class="shard-icon" [matMenuTriggerFor]="shardMenu">
               <clr-tooltip>
                 <clr-icon clrTooltipTrigger shape="layers" size="24"  class="is-solid"></clr-icon>
-                <clr-tooltip-content clrPosition="right"  clrSize="xs" *clrIfOpen>
+                <clr-tooltip-content clrPosition="left"  clrSize="xs" *clrIfOpen>
```

Comment:
Not familiar with the UI here, so I have a curious question: what would the task look like if a task is both call-cached and scattered? My guess is that the "scatter" icon will sit to the left?

Reply:
I believe a scattered task cannot be call-cached, but I might be wrong. @ruchim Can you chime in on this one?

Reply:
Per the offline discussion, this shouldn't be a problem, and we can also update it in a separate PR!